httplib.h

```diff
@@ -1865,7 +1869,7 @@ class stream_line_reader {
 namespace detail {
 
 inline bool is_hex(char c, int &v) {
-  if (0x20 <= c && isdigit(c)) {
+  if (0x20 <= c && (isdigit(c) != 0)) {
```

Comment:
Checking that the return value of `isdigit()` is not zero decreases readability - the fact that the function returns `int` is almost irrelevant, as the function returns "Non-zero if the character is a numeric character, zero otherwise", i.e. it returns a boolean. The return type is `int` probably because when `isdigit()` was added to the C standard library `_Bool` wasn't even a thing.
By the way, you might want to change this to [`std::isdigit()`](https://en.cppreference.com/w/cpp/string/byte/isdigit)